sections/futures/ProfitCalculator/ProfitCalculator.tsx

```diff
@@ -0,0 +1,225 @@
+import { PositionSide } from '../types';
+
+const ProfitCalculator = ({ marketAsset, marketAssetRate, setOpenProfitCalcModal }: any) => {
+	const marketAsset__RemovedSChar = marketAsset[0] === 's' ? marketAsset.slice(1) : marketAsset;
+	const { t } = useTranslation();
+
+	// Wei
+	const [entryPrice, setEntryPrice] = useState<string>('');
+	const [exitPrice, setExitPrice] = useState<string>('');
+	const [gainPercent, setGainPercent] = useState<string>('');
+	const [stopLoss, setStopLoss] = useState<string>('');
+	const [lossPercent, setLossPercent] = useState<string>('');
+	const [marketAssetPositionSize, setMarketAssetPositionSize] = useState<string>('');
+	const [basePositionSize, setBasePositionSize] = useState<string>('');
+	// Custom type
+	const [leverageSide, setLeverageSide] = useState<PositionSide>(PositionSide.LONG);
+
+	const onEntryPriceAmountChange = (value: string) => {
+		setEntryPrice(value);
+	};
+
+	const onExitPriceAmountChange = (value: string, fromLeverage: boolean = false) => {
+		setExitPrice(fromLeverage ? (value === '' ? '' : wei(value).toNumber().toFixed(2)) : value);
+		setGainPercent(
+			value === '' ? '' : (100 * (wei(value).div(entryPrice).toNumber() - 1)).toFixed(2)
+		);
+	};
+
+	const onGainPercentChange = useCallback(
+		(value: string) => {
+			setGainPercent(value);
+			setExitPrice(
+				value === '' ? '' : wei(value).div(100).add(1).mul(entryPrice).toNumber().toFixed(2)
+			);
+		},
+		[entryPrice]
+	);
+
+	const onStopLossAmountChange = (value: string, fromLeverage: boolean = false) => {
+		setStopLoss(fromLeverage ? (value === '' ? '' : wei(value).toNumber().toFixed(2)) : value);
+		setLossPercent(
+			value === '' ? '' : ((1 - wei(value).div(entryPrice).toNumber()) * 100).toFixed(2)
+		);
+	};
+
+	const onLossPercentChange = useCallback(
+		(value: string) => {
+			setLossPercent(value);
+			setStopLoss(
+				value === '' ? '' : wei(1).sub(wei(value).div(100)).mul(entryPrice).toNumber().toFixed(2)
+			);
+		},
+		[entryPrice]
+	);
+
+	const onTradeAmountChange = useCallback(
+		(value: any, fromLeverage: boolean = false) => {
+			setMarketAssetPositionSize(
+				fromLeverage ? (value === '' ? '' : wei(value).toNumber().toFixed(2)) : value
+			);
+			setBasePositionSize(
+				value === '' ? '' : (parseFloat(entryPrice) * parseFloat(value)).toFixed(2)
+			);
+		},
+		[entryPrice]
+	);
+
+	const onTradeAmountSUSDChange = useCallback(
+		(value: string) => {
+			setBasePositionSize(value);
+			setMarketAssetPositionSize(
+				value === '' ? '' : wei(value).div(entryPrice).toNumber().toFixed(2)
+			);
+		},
+		[entryPrice]
+	);
+
+	useEffect(() => {
+		setEntryPrice(marketAssetRate);
+	}, []);
+
+	return (
+		<>
+			<BaseModal
+				onDismiss={() => setOpenProfitCalcModal(false)}
+				isOpen={true}
+				/**
+				 * @todo IDK how to make this comply with this project's style, e.g.
+				 *       `t('modals.confirm-transaction.title')`
+				 */
+				title={t('Profit Calculator')}
```

Comment:
See the `en.json` file. cc @Tburm

Reply:
@0xwontonsoup The file is located at `/translations/en.json`, you can see how we're pulling individual translations there. You can add new entries for the calculator in `futures`